### PR TITLE
Iterate and remove nodes without container copy

### PR DESCRIPTION
**Pull request by promag**
Removes the need for`vNodesCopy` and `vNodesDisconnectedCopy`.

### CODE REVIEW

File: src/net.cpp [hunk 1]
```diff
@@ -1171,14 +1169,15 @@ void CConnman::ThreadSocketHandler()
                     // hold in disconnected pool until all refs are released
                     pnode->Release();
                     vNodesDisconnected.push_back(pnode);
+                    continue;
                 }
+                it++;
```
Comment (laanwj) [inline src/net.cpp]: Nit: ++it? (make no code difference probably, but I think it's good practice, and avoids later patches changing it around)

Comment (promag) [reply src/net.cpp]: Ok.

**Comment by theuni** [comment]
This code goes away with #11227, which is in the process of being PR'd in batches. I have no objection to improvements here, but I'd prefer to avoid making changes that will only cause conflicts with deleted code.

**Comment by laanwj** [comment]
> This code goes away with #11227, which is in the process of being PR'd in batches. I have no objection to improvements here, but I'd prefer to avoid making changes that will only cause conflicts with deleted code.

Good point, the review time is better spent elsewhere then.

**Comment by promag** [comment]
In that case.

**Pull request closed without merging**